### PR TITLE
docs: update

### PR DESCRIPTION
**Pull request by tjdevries**
@bfredl @vigoux

### CODE REVIEW

**Review by justinmk — APPROVED** [review]

File: runtime/doc/api.txt [hunk 1]
```diff
@@ -537,6 +537,12 @@ nvim__put_attr({id}, {start_row}, {start_col}, {end_row}, {end_col})
                 should probably be derived from a reformed bufhl/virttext
                 interface with full support for multi-line ranges etc
 
+nvim__screenshot({path})                                  *nvim__screenshot()*
```
Comment (bfredl) [inline runtime/doc/api.txt:540]: not in this PR, but later it should be made explicit automagically that every `nvim__` is unstable.

File: runtime/doc/api.txt [hunk 2]
```diff
@@ -1974,22 +1956,35 @@ nvim_buf_line_count({buffer})                          *nvim_buf_line_count()*
...
+
                 Parameters: ~
                     {buffer}  Buffer handle, or 0 for current buffer
                     {ns_id}   Namespace id from |nvim_create_namespace()|
-                    {id}      Extmark id, or 0 to create new
                     {line}    Line number where to place the mark
                     {col}     Column where to place the mark
-                    {opts}    Optional parameters. Currently not used.
+                    {opts}    Optional parameters.
+                              • id : id of the extmark to edit.
```
Comment (bfredl) [inline runtime/doc/api.txt:1980]: "edit" is a bit weird. maybe "replace" or "update" (change in C code :).

**Review by bfredl — APPROVED** [review]

**Review by vigoux — APPROVED** [review]

**Comment by vigoux** [comment]
Little thought : I just saw that `lsp` has a dedicated doc file, and I really think `treesitter` should get the same, this way we can add a bit of doc on how to write queries and so on.

**Comment by tjdevries** [comment]
I can do the other stuff later. We should make sure docs get updated with breaking changes.

**Pull request merged**